tests/queries/0_stateless/03095_window_functions_qualify.sql

```diff
@@ -0,0 +1,19 @@
+SET allow_experimental_analyzer = 1;
+
+SELECT number, COUNT() OVER (PARTITION BY number % 3) AS partition_count FROM numbers(10) QUALIFY partition_count = 4 ORDER BY number;
```

Comment:
Let's also add a test with a query like:
```sql
SELECT
  number,
  COUNT() OVER (PARTITION BY number % 3) AS partition_count
FROM numbers(10)
QUALIFY COUNT() OVER (PARTITION BY number % 3) = 4
ORDER BY number;
```
and check that window function is computed only once.

Reply:
Updated.